packages/docs/stories/components/checkbox.stories.ts

```diff
@@ -0,0 +1,140 @@
+import type { Meta, StoryObj } from '@storybook/web-components';
+import { html } from 'lit';
+import { userEvent } from '@storybook/testing-library';
+import { waitUntil } from '@open-wc/testing-helpers';
+import docsTokens from '../../../tokens/src/figma-tokens/_docs.json';
+import { storybookDefaults, storybookHelpers, storybookTemplate } from '../../src/helpers/component.js';
+const { args, argTypes } = storybookDefaults('syn-checkbox');
+const { overrideArgs } = storybookHelpers('syn-checkbox');
+const { generateTemplate } = storybookTemplate('syn-checkbox');
+
+const meta: Meta = {
+  component: 'checkbox',
+  args,
+  argTypes,
+  title: 'Components/syn-checkbox',
+  parameters: {
+    docs: {
+      description: {
+        component: docsTokens?.components?.['checkbox']?.default?.description?.value ?? 'No Description',
+      },
+    }
+  }
+};
+export default meta;
+
+type Story = StoryObj;
+
+export const Default = {
+  render: (args: any) => {
+    return generateTemplate({ args });
+  },
+  parameters: {
+    docs: {
+      description: {
+        story: docsTokens?.components?.['checkbox']?.default?.description?.value ?? 'No Description',
+      }
+    }
+  }
+} as Story;
+
+
+/**
+ * Use the checked attribute to activate the checkbox.
+ */
+export const Checked: Story = {
+  render: () => html`<syn-checkbox checked>Checked</syn-checkbox>`,
+};
+
+/**
+ * Use the indeterminate attribute to make the checkbox indeterminate.
+ */
+export const Indeterminate: Story = {
+  render: () => html`<syn-checkbox indeterminate>Indeterminate</syn-checkbox>`,
+};
+
+/**
+ * Use the disabled attribute to disable the checkbox.
+ */
+export const Disabled: Story = {
+  render: () => html`<syn-checkbox disabled>Disabled</syn-checkbox>`,
+};
+
+/**
+ * Use the size attribute to change a checkbox's size.
+ */
+export const Sizes: Story = {
+  render: () => html`<syn-checkbox size="small">Small</syn-checkbox>
+<br />
+<syn-checkbox size="medium">Medium</syn-checkbox>
+<br />
+<syn-checkbox size="large">Large</syn-checkbox>`,
+};
+
+/**
+ * Use the setCustomValidity() method to set a custom validation message. This will prevent the form from submitting and make the browser display the error message you provide. To clear the error, call this function with an empty string.
+ */
+export const CustomValidity: Story = {
+  play: async ({ canvasElement }) => {
+    try {
+      const checkbox = canvasElement.querySelector('syn-checkbox');
+      const button = canvasElement.querySelector('button');
+
+      await waitUntil(() => checkbox?.shadowRoot?.querySelector('input'));
+
+      if (button) {
+        await userEvent.click(button);
+      }
+    } catch (error) {
+      console.error('Error in play function:', error);
+    }
+  },
+  render: () => html`
+    <form class="custom-validity">
+      <syn-checkbox>Check me</syn-checkbox>
+      <br />
+      <button size="medium" type="submit">Submit</button>
```

Comment:
```suggestion
      <syn-button size="medium" type="submit">Submit</syn-button>
```

I would use the syn-button here.

Reply:
This seems to be harder to do. I would like to create a new ticket out of this as it will also have to be adjusted for the input stories. I did not get it to work correctly in the first place. I have a small hint where this is coming from, but would also need to update some internal packages for it to work (e.g. dom-testing-library, openwc-testing and others). Seems like a bigger problem to me.